custom-metrics-stackdriver-adapter/pkg/adapter/provider/provider.go

```diff
@@ -83,14 +86,32 @@ func NewStackdriverProvider(kubeClient *corev1.CoreV1Client, mapper apimeta.REST
 	}
 }
 
+// GetMetricByName fetches a particular metric for a particular object.
+// The namespace will be empty if the metric is root-scoped.
+func (p *StackdriverProvider) GetMetricByName(name types.NamespacedName, info provider.CustomMetricInfo) (*custom_metrics.MetricValue, error) {
+	if name.Namespace == "" {
+		return p.GetRootScopedMetricByName(info.GroupResource, name.Name, info.Metric)
+	}
+	return p.GetNamespacedMetricByName(info.GroupResource, name.Namespace, name.Name, info.Metric)
+}
+
+// GetMetricBySelector fetches a particular metric for a set of objects matching
+// the given label selector.  The namespace will be empty if the metric is root-scoped.
+func (p *StackdriverProvider) GetMetricBySelector(namespace string, selector labels.Selector, info provider.CustomMetricInfo) (*custom_metrics.MetricValueList, error) {
+	if namespace == "" {
+		return p.GetRootScopedMetricBySelector(info.GroupResource, selector, info.Metric)
+	}
+	return p.GetNamespacedMetricBySelector(info.GroupResource, namespace, selector, info.Metric)
+}
+
 // GetRootScopedMetricByName queries Stackdriver for metrics identified by name and not associated
 // with any namespace. Current implementation doesn't support root scoped metrics.
 func (p *StackdriverProvider) GetRootScopedMetricByName(groupResource schema.GroupResource, name string, escapedMetricName string) (*custom_metrics.MetricValue, error) {
```

Comment:
Did the framework switched from using those to only "GetMetricByName" and "GetMetricBySelector"? Can you limit the visibility of this older functions?

Reply:
Yes, it did. Fixed.